pulsar/client_impl.go

```diff
@@ -66,8 +66,10 @@ func newClient(options ClientOptions) (Client, error) {
 	if connectionMaxIdleTime == 0 {
 		connectionMaxIdleTime = defaultConnMaxIdleTime
 	} else if connectionMaxIdleTime > 0 && connectionMaxIdleTime < minConnMaxIdleTime {
-		return nil, newError(InvalidConfiguration, fmt.Sprintf("Connection max idle time should be at least %f "+
-			"seconds", minConnMaxIdleTime.Seconds()))
+		logger.Warnf("Connection idle detect interval seconds default same as max idle seconds, but max idle"+
+			" seconds less than %d, to avoid checking"+
+			" connection status too much, use default value : %d", minConnMaxIdleTime.Minutes(), minConnMaxIdleTime.Minutes())
+		connectionMaxIdleTime = minConnMaxIdleTime
```

Comment:
This is a misconfiguration. And IMO, we should throw an error for it instead of the warning. Otherwise, the client's behavior may differ from what users expect.
Are there any cases where this change is needed?

Reply:
Thank you for your explanatition, it as is might be a good choice as weel.